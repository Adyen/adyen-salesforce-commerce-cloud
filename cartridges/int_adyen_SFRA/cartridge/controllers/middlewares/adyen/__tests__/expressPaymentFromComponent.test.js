"use strict";

/* eslint-disable global-require */
var expressPaymentFromComponent;
var res;
var req;
beforeEach(function () {
  var _require = require('../../index'),
    adyen = _require.adyen;
  expressPaymentFromComponent = adyen.expressPaymentFromComponent;
  jest.clearAllMocks();
  req = {
    currentCustomer: {
      profile: {
        customerNo: '12321'
      },
      addressBook: {
        preferredAddress: {
          countryCode: {}
        }
      }
    },
    form: {
      data: {
        paymentMethod: {
          type: 'mocked_type'
        }
      }
    }
  };
  res = {
    redirect: jest.fn(),
    json: jest.fn()
  };
});
afterEach(function () {
  jest.resetModules();
});
describe('Express Payment from Component', function () {
  it('should fail without default address', function () {
    delete req.currentCustomer.addressBook.preferredAddress;
    req.form.data = JSON.stringify(req.form.data);
    expressPaymentFromComponent(req, res, jest.fn());
    expect(res.json.mock.calls).toMatchSnapshot();
  });
  it('should cancel transaction', function () {
    req.form.data.cancelTransaction = true;
    req.form.data = JSON.stringify(req.form.data);
    expressPaymentFromComponent(req, res, jest.fn());
    expect(res.json.mock.calls).toMatchSnapshot();
  });
  it('should return json response', function () {
    req.form.data = JSON.stringify(req.form.data);
    expressPaymentFromComponent(req, res, jest.fn());
    expect(res.json.mock.calls).toMatchSnapshot();
  });
});