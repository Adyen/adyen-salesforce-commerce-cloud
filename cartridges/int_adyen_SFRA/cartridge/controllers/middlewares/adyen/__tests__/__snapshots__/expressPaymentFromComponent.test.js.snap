// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Express Payment from Component should cancel transaction 1`] = `
Array [
  Array [
    Object {},
  ],
]
`;

exports[`Express Payment from Component should fail without default address 1`] = `
Array [
  Array [
    Object {
      "error": true,
    },
  ],
]
`;

exports[`Express Payment from Component should return json response 1`] = `
Array [
  Array [
    Object {
      "orderNo": "mocked_orderNo",
      "resultCode": "Authorised",
    },
  ],
]
`;
